common/lib/transport/connectionmanager.js

```diff
@@ -647,8 +648,24 @@ var ConnectionManager = (function() {
 		if((wasActive && noTransportsScheduledForActivation) ||
 			(wasActive && (state === 'failed') || (state === 'closed')) ||
 			(currentProtocol === null && wasPending && this.pendingTransports.length === 0)) {
+			/* If we're disconnected with a 5xx we need to try fallback hosts
+			 * (RTN14d), but (a) due to how the upgrade sequence works, the
+			 * host/transport selection sequence only cares about getting to
+			 * `preconnect` (eg establishing a websocket) getting a `disconnected`
+			 * protocol message afterwards is too late; and (b) host retry only
+			 * applies to connectBase unless the stored preference transport doesn't
+			 * work. We solve this by unpersisting the transport preference and
+			 * setting an instance variable to force fallback hosts to be used (if
+			 * any) here. Bit of a kludge, but no real better alternatives without
+			 * rewriting the entire thing */
+			if(state === 'disconnected' && error && error.statusCode > 500) {
+				this.unpersistTransportPreference();
+				this.forceFallbackHost = true;
+			}
 			/* TODO remove below line once realtime sends token errors as DISCONNECTEDs */
```

Comment:
Does this still need to be done?

Reply:
Yup, that's waiting on https://github.com/ably/realtime/issues/1154

Reply:
actually, wait, no, that's a different thing. Not sure, will check.